src/geff/io/read.py

```diff
@@ -120,8 +111,20 @@ def get_construct_func(backend: SupportedBackend) -> ConstructFunc[Any]:
     match backend:
         case SupportedBackend.NETWORKX:
             return construct_nx
-        case SupportedBackend.GRAPH_DICT:
-            return construct_identity
+        case SupportedBackend.RUSTWORKX:
+            if construct_rx is None:
+                raise ImportError(
+                    "rustworkx is not installed. To read a GEFF to the rustworkx backend please "
+                    "use `pip install 'geff[rx]'`"
+                )
+            return construct_rx
+        case SupportedBackend.SPATIAL_GRAPH:
+            if construct_sg is None:
+                raise ImportError(
+                    "spatial-graph is not installed. To read a GEFF to the spatial-graph backend "
+                    "please use `pip install 'geff[spatial-graph]'`"
+                )
+            return construct_sg
```

Comment:
since both of these `io` modules already raise a similar error, what do you think of just:


```suggestion
        case SupportedBackend.RUSTWORKX:
            from geff.rustworkx.io import construct_rx

            return construct_rx
        case SupportedBackend.SPATIAL_GRAPH:
            from geff.spatial_graph.io import construct_sg

            return construct_sg
```


and them remove the `find_spec` logic above?

Reply:
I did consider this but I didn't want to add import time as an overhead to the read func, but tbh I don't know anything about how python imports work under the hood and if they are cached, if you don't think it's a problem then yeah it looks neater

Reply:
Oh yeah: I'd say that's premature optimization:)

Once imported, a reimport is simply a dictionary lookup.  Way less code, microsecond lookup

Reply:
And also; wouldn't you rather only pay that import time cost (once) when you're actually using that backend, and not _every_ time you import geff?

Reply:
makes sense! I will update